### ISSUE TITLE
Add feature g:disable_when_inserting

### ISSUE THREAD

**Issue by baabelfish** [w0rp/ale#420]
It would be nice to have a built in flag for disabling linting while user is inserting test (in insert mode). I don't really like watching red flags/text while typing.

**Comment by w0rp** [comment]
This is already supported. See `:help g:ale_lint_on_text_changed`.

**Comment by w0rp** [comment]
See also `:help g:ale_lint_delay`. You can increase the delay after which linting will be done. The behaviour is described in the help for the previous option.

**Comment by baabelfish** [comment]
``g:ale_lint_on_text_changed`` disables the linting completely. I would like to disable it in insert mode only. Setting delay is not really an option because I might take pauses longer than the delay while writing new code and I would like the delay to be really low when editing in normal mode.

**Comment by w0rp** [comment]
When do you want linting to be run? When text is changed, but not through insert mode? Perhaps only after exiting insert mode? When a file is saved?

**Comment by baabelfish** [comment]
Indeed when exiting insert mode and in normal mode where most of the magic happens ;) Usually when inserting a new line of code I have to watch red text until I write ';' which is annoying.
